pkg/stanza/fileconsumer/internal/tracker/tracker.go

```diff
@@ -110,9 +116,16 @@ func (t *fileTracker) ClosePreviousFiles() (filesClosed int) {
 	return
 }
 
+func (t *fileTracker) SetPersister(persister operator.Persister) {
```

Comment:
Let's pass this in during instantiation so that it is not an effect that can be applied later.

Reply:
Also because it may not be a concern of other implementation of tracker.

Reply:
@djaglowski we instantiate the tracker while building the config. The `persister` is not available to us in that part of code. It's available in `Start(...)`.